### PR TITLE
Fix hang in batchnorm training mode, issue #30279

### PR DESCRIPTION
**Pull request by iwroszTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/30279

### Problem description
Batch norm operation hangs for large input tensors, when ran in training mode.

### What's changed
Handling of several circular buffers was fixed in the running_statistics_sfpu kernel

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [x] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [x] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/running_statistics_sfpu_kernel.cpp [hunk 1]
```diff
@@ -173,54 +167,47 @@ void MAIN {
...
+            tile_regs_commit();
+            pack_tile(tile_index * 2, cb_updated_running_var);
             tile_regs_release();
             cb_push_back(cb_updated_running_var, onetile);
+
             cb_pop_front(cb_tmp3, onetile);
             cb_pop_front(cb_tmp2, onetile);
         }
+
+        cb_push_back(cb_out0, 1);
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/running_statistics_sfpu_kernel.cpp:214]: I think the idea was that the cb_out0 would get the same value as either running var. Also, usually there should be a cb_reserve_back to get the indexing right for writing.

Probably easiest would be to add:
```
cb_reserve_back(cb_out0, 1);
...
pack_tile(tile_index * 2, cb_updated_running_mean);
pack_tile(tile_index * 2, cb_out0);
...
pack_tile(tile_index * 2, cb_updated_running_var);
pack_tile(tile_index * 2, cb_out0);
...
```
One of the two if statements should be taken.

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp [hunk 2]
```diff
@@ -57,7 +57,9 @@ Tensor BatchNorm::invoke(
         batch_mean = mean_NHW(input, memory_config, compute_kernel_config);
         auto mean_sq = mean_NHW(ttnn::square(input, memory_config), memory_config, compute_kernel_config);
         batch_var = ttnn::subtract(mean_sq, ttnn::square(batch_mean, memory_config), std::nullopt, memory_config);
-        Tensor stats = ttnn::prim::running_statistics(
+        // The return Tensor of running_statistics is only for infra purposes.
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp]: Given the issue with cb_out0 in the compute kernel, please test if the suggested changes make the output be sensible. If yes, please remove this comment.

Comment (iwroszTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp]: I've verified that with the latest patch the output tensor holds either the running variance stat, or the running mean stat (when the var tensor is not provided).

**Review by bbradelTT — APPROVED** [review]
